python-checks/src/main/java/org/sonar/python/checks/CollapsibleIfStatementsCheck.java

```diff
@@ -58,11 +63,36 @@ public void visitIfStatement(IfStatement ifStatement) {
       && statements.size() == 1
       && statements.get(0).is(Tree.Kind.IF_STMT)) {
       IfStatement singleIfChild = (IfStatement) statements.get(0);
-      if (singleIfChild.isElif() || singleIfChild.elseBranch() != null || !singleIfChild.elifBranches().isEmpty()) {
+      if (isException(singleIfChild, ifStatement)) {
         return;
       }
       addIssue(singleIfChild.keyword(), MESSAGE).secondary(ifStatement.keyword(), "enclosing");
     }
     super.visitIfStatement(ifStatement);
   }
+
+  private static boolean isException(IfStatement singleIfChild, IfStatement enclosingIfStatement) {
+    return singleIfChild.isElif()
+      || singleIfChild.elseBranch() != null
+      || !singleIfChild.elifBranches().isEmpty()
+      || wouldCauseLongLineLength(singleIfChild, enclosingIfStatement)
+      || singleIfChild.condition().is(ASSIGNMENT_EXPRESSION)
+      || enclosingIfStatement.condition().is(ASSIGNMENT_EXPRESSION)
+      || hasCommentsBetweenEnclosingAndChildIf(singleIfChild, enclosingIfStatement);
+  }
+
+  private static boolean hasCommentsBetweenEnclosingAndChildIf(IfStatement singleIfChild, IfStatement enclosingIfStatement) {
+    return TreeUtils.tokens(enclosingIfStatement).stream()
+      .anyMatch(token -> !token.trivia().isEmpty() && token.trivia().get(0).token().line() < singleIfChild.firstToken().line());
+  }
+
+  private static boolean wouldCauseLongLineLength(IfStatement singleIfChild, IfStatement enclosingIf) {
```

Comment:
It's a bit nitpicky but I think we should also account for the fact that, on a single line, an `and` as well as 2 additional spaces surrounding it would be required as well.
Currently, rewriting the following snippet with 2 nested ifs raises an issue, despite the line being 84 characters long:
```
if blaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa and bluuuuuu:
  ...

```